scripts/run.sh

```diff
@@ -1,16 +1,160 @@
+	# wp.config.php
+	printf "=> Generating wp.config.php file... "
+	sudo -u www-data wp core config >/dev/null 2>&1 || ERROR $LINENO "Could not generate wp-config.php file"
+	printf "Done!\n"
+
+else
+	printf "=> wp-config.php exists. SKIPPING...\n"
+fi
+
+# Install Database
+if [ ! "$(wp core is-installed --allow-root >/dev/null 2>&1 && echo $?)" ]; then
+
+	printf "=> Creating database '%s'... " "$DB_NAME"
+	sudo -u www-data wp db create >/dev/null 2>&1 || ERROR $LINENO "Database creation failed"
+	printf "Done!\n"
+
+	# If an SQL file exists in /data => load it
+	if [ "$(stat -t /data/*.sql >/dev/null 2>&1 && echo $?)" ]; then
+
+		DATA_PATH=$(find /data/*.sql | head -n 1)
+		printf "=> Loading data backup from %s... " "$DATA_PATH"
+		sudo -u www-data wp db import "$DATA_PATH" >/dev/null 2>&1 || ERROR $LINENO "Could not import database"
+		printf "Done!\n"
+
+		# If SEARCH_REPLACE is set => Replace URLs
+		if [ "$SEARCH_REPLACE" != false ]; then
+
+			printf "=> Replacing URLs... "
+			REPLACEMENTS=$(sudo -u www-data wp search-replace "$BEFORE_URL" "$AFTER_URL" \
+				--no-quiet --skip-columns=guid | grep replacement) || \
+				ERROR $((LINENO-2)) "Could not execute SEARCH_REPLACE on database"
+			echo -ne "$REPLACEMENTS\n"
+
+		fi
+
+	else
+		printf "%s\n" \
+			"=> No database backup found. Initializing new database..." " " \
+			"	  DEFAULT DATABASE CREDENTIALS" \
+			"=========================================" \
+			" Site URL:	   $([ "$AFTER_URL" ] && echo "$AFTER_URL" || echo localhost:8080)" \
+			" Site Title:	 $DB_NAME" \
+			" Admin Username: root" \
+			" Admin Password: $DB_PASS" \
+			" Admin Email:	admin@${DB_NAME}.com" \
+			"=========================================" " "
+
+		sudo -u www-data wp core install >/dev/null 2>&1 || ERROR $LINENO "WordPress Install Failed"
+	fi
+
+	printf "=> Database initialization completed sucessfully!\n"
+
+else
+	printf "=> Database '%s' exists. SKIPPING...\n" "$DB_NAME"
+fi
+
+
+# .htaccess
+if [ ! -f /app/.htaccess ]; then
+	printf "=> Generating .htaccess file... "
+	sudo -u www-data wp rewrite flush --hard >/dev/null 2>&1 || ERROR $LINENO "Could not generate .htaccess file"
+	printf "Done!\n"
 else
-  echo "Wordpress already setup!"
+	printf "=> .htaccess exists. SKIPPING...\n"
 fi
 
-if [ ! -f /app/.mysql_db_created ]; then
-  /scripts/setup-database.sh
+
+# Adjust Filesystem Permissions
+printf "=> Adjusting filesystem permissions... "
+groupadd -f docker && usermod -aG docker www-data
+find /app -type d -exec chmod 755 {} \;
+find /app -type f -exec chmod 644 {} \;
+chmod -R 775 /app/wp-content/uploads \
+	&& chown -R :docker /app/wp-content/uploads
+printf "Done!\n"
+
+# Install Plugins
+if [ "$PLUGINS" ]; then
+	printf "=> Checking plugins...\n"
+	while IFS=',' read -ra plugin; do
+		for i in "${!plugin[@]}"; do
+			sudo -u www-data wp plugin is-installed "${plugin[$i]}"
+			if [ $? -eq 0 ]; then
+				printf "=> ($((i+1))/${#plugin[@]}) Plugin '%s' found. SKIPPING...\n" "${plugin[$i]}"
```

Comment:
I agree that explicit dependency update is a better pattern. But we are not providing a way to define which version you need, so on installation, it always get the latest one. That's why it feels natural to always keep the stack up to date at the latest version. That said, I think we can keep this discussion for another PR.